httpd/filters.go

```diff
@@ -0,0 +1,147 @@
+package httpd
```

Comment:
I'm not sure I would have created this file. `handler.go` is only ~300 lines long. I'm not sure much was gained by pulling this code out into its own file. This code is still handler code -- it knows intricate details about the handler object.